sale_coupon_product_management/models/sale_coupon_program_option.py

```diff
@@ -98,7 +98,14 @@ def _get_related_record(self, program, path):
             return program
         return operator.attrgetter(path)(program)
 
-    def get_program_values(self):
+    def _update_values_from_program(self, opt_vals, program):
+        self.ensure_one()
+        if self.option_type == "discount_fixed_amount":
+            opt_vals[DISCOUNT_PRODUCT_FNAME][
+                "list_price"
+            ] = program.discount_fixed_amount
```

Comment:
As asked by @mmequignon don't change values of `opt_vals` directly into the function.
Return the new values and update the dict out of the function.